addons/account/models/account_move.py

```diff
@@ -1480,9 +1475,38 @@ def _track_subtype(self, init_values):
     # -------------------------------------------------------------------------
 
     @api.multi
-    def _is_invoice(self):
-        self.ensure_one()
-        return self.type in ('out_invoice', 'out_refund', 'in_invoice', 'in_refund', 'out_receipt', 'in_receipt')
+    def _entry_belong_to_subset(self, tested_type, types):
+        return self.type in types if tested_type is None else tested_type in types
+
+    def is_invoice(self, tested_type=None, include_receipts=False):
+        types = ['out_invoice', 'out_refund', 'in_refund', 'in_invoice']
+        if include_receipts:
+            types += ['out_receipt', 'in_receipt']
+        return self._entry_belong_to_subset(tested_type, types=types)
+
+    def is_sale_document(self, tested_type=None, include_receipts=False):
+        types = ['out_invoice', 'out_refund']
+        if include_receipts:
+            types += ['out_receipt']
+        return self._entry_belong_to_subset(tested_type, types=types)
+
+    def is_purchase_document(self, tested_type=None, include_receipts=False):
+        types = ['in_refund', 'in_invoice']
+        if include_receipts:
+            types += ['in_receipt']
+        return self._entry_belong_to_subset(tested_type, types=types)
+
+    def is_inbound(self, tested_type=None, include_receipts=True):
+        types = ['out_invoice', 'in_refund']
+        if include_receipts:
+            types += ['out_receipt']
+        return self._entry_belong_to_subset(tested_type, types=types)
+
+    def is_outbound(self, tested_type=None, include_receipts=True):
+        types = ['out_refund', 'in_invoice']
+        if include_receipts:
+            types += ['in_receipt']
+        return self._entry_belong_to_subset(tested_type, types=types)
```

Comment:
to rewrite like `is_invoice`
Manage `tested_type` case by case with `move_type in ('out_refund', 'in_invoice')` because It seems less ugly to me